gensim/models/poincare.py

```diff
@@ -731,27 +739,457 @@ def compute_loss(self):
+
+        """
+        vector_1 = self.word_vec(term_1)
+        vector_2 = self.word_vec(term_2)
+        return self.poincare_distance(vector_1, vector_2)
+
+    def similarity(self, term_1, term_2):
+        """
+        Return similarity based on Poincare distance between two terms.
+
+        Parameters
+        ----------
+        term_1 : str
+            Input term.
+        term_2 : str
+            Input term.
+
+        Returns
+        -------
+        float
+            Similarity between the vectors for `term_1` and `term_2` (between 0 and 1).
+
+        Examples
+        --------
+
+        >>> model.similarity('mammal.n.01', 'carnivore.n.01')
+        0.73
+
+        Notes
+        -----
+        Raises KeyError if either of `term_1` and `term_2` is absent from vocab.
+        Similarity lies between 0 and 1.
+
+        """
+        return 1 / (1 + self.distance(term_1, term_2))
+
+    def most_similar(self, word_or_vector, topn=10, restrict_vocab=None):
+        """
+        Find the top-N most similar words to the given word or vector, sorted in increasing order of distance.
+
+        Parameters
+        ----------
+
+        word_or_vector : str or numpy.array
+            word or vector for which similar words are to be found.
+        topn : int or None, optional
+            number of similar words to return, if `None`, returns all.
+        restrict_vocab : int or None, optional
+            Optional integer which limits the range of vectors which are searched for most-similar values.
+            For example, restrict_vocab=10000 would only check the first 10000 word vectors in the vocabulary order.
+            This may be meaningful if vocabulary is sorted by descending frequency.
+
+        Returns
+        --------
+        list of tuples (str, float)
+            List of tuples containing (word, distance) pairs in increasing order of distance.
+
+        Examples
+        --------
+        >>> vectors.most_similar('lion.n.01')
+        [('lion_cub.n.01', 0.4484), ('lionet.n.01', 0.6552), ...]
+
+        """
+        if not restrict_vocab:
+            all_distances = self.distances(word_or_vector)
+        else:
+            words_to_use = self.index2word[:restrict_vocab]
+            all_distances = self.distances(word_or_vector, words_to_use)
+
+        if isinstance(word_or_vector, string_types):
+            word_index = self.vocab[word_or_vector].index
+        else:
+            word_index = None
+        if not topn:
+            closest_indices = matutils.argsort(all_distances)
+        else:
+            closest_indices = matutils.argsort(all_distances, topn=1 + topn)
+        result = [
+            (self.index2word[index], float(all_distances[index]))
+            for index in closest_indices if (not word_index or index != word_index)  # ignore the input word
+        ]
+        if topn:
+            result = result[:topn]
+        return result
+
+    def distances(self, word_or_vector, words_2=[]):
```

Comment:
It's a bad practice to use mutable types as default parameter values. `words_2=()` better.

Reply:
Fixed